cmd/cerbosctl/store/export/internal/exporter.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/cerbos/cerbos/internal/util"
+)
+
+type Exporter struct {
+	gzipWriter *gzip.Writer
+	tarWriter  *tar.Writer
+	zipWriter  *zip.Writer
+	file       *os.File
+	path       string
+}
+
+func (e *Exporter) WriteJSON(name string, jsonData []byte) error {
+	return e.write(name, jsonData)
+}
+
+func (e *Exporter) WriteYAML(name string, jsonData []byte) error {
+	data, err := yaml.JSONToYAML(jsonData)
+	if err != nil {
+		return fmt.Errorf("failed to convert JSON to YAML: %w", err)
+	}
+
+	return e.write(name, data)
+}
+
+func (e *Exporter) write(name string, data []byte) error {
+	var w io.Writer
+	var err error
+	switch {
+	case util.IsTar(e.path) || util.IsGzip(e.path):
```

Comment:
You know what kind of file this is at the point you create the exporter using `NewExporter`. So, you could store that information in the struct as a bool and check that instead of performing this name check over and over again for each file you are adding.

I'd probably design this in a different way though. I'd make `Exporter` an interface and then have three different implementation of it for tar, zip and dir. When `NewExporter` is called, based on the output filename, I'd return the correct implementation of `Exporter` back to the client. That way the code for each output format is nicely separated and you don't need to wade through a bunch of if statements.